### PR TITLE
Dataset beta

### PR DESCRIPTION
**Pull request by lfayoux**
Adds functionality to use the dataset API for select beta customers. Allows embed jobs and cluster to accept dataset ids.

### CODE REVIEW

File: cohere/client_async.py [hunk 1]
```diff
@@ -478,14 +484,126 @@ async def rerank(
...
+        embeddings_url: str = None,
         min_cluster_size: Optional[int] = None,
         n_neighbors: Optional[int] = None,
         is_deterministic: Optional[bool] = None,
         generate_descriptions: Optional[bool] = None,
-    ) -> AsyncCreateClusterJobResponse:
+    ) -> AsyncClusterJobResult:
         """Create clustering job.
 
         Args:
```
Comment (daniel-cohere) [inline cohere/client_async.py:609]: `input_dataset_id` should have an arg comment here?

File: cohere/client_async.py [hunk 2]
```diff
@@ -574,82 +694,94 @@ async def wait_for_cluster_job(
...
         compression_codebook: Optional[str] = None,
         text_field: Optional[str] = None,
-        output_format: Optional[str] = None,
-    ) -> AsyncCreateBulkEmbedJobResponse:
-        """Create bulk embed job.
+    ) -> AsyncEmbedJob:
+        """Create embed job.
 
         Args:
             input_file_url (str): File with texts to embed.
```
Comment (daniel-cohere) [inline cohere/client_async.py:710]: this should change to `input_dataset`?

File: tests/async/test_async_dataset.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+import pytest
+
+from cohere import AsyncClient
+from cohere.responses import AsyncDataset
+
+
+@pytest.mark.asyncio
+async def test_async_create_dataset(async_client: AsyncClient):
+    job = await async_client.create_dataset(
```
Comment (daniel-cohere) [inline tests/async/test_async_dataset.py:15]: same as below `job`/`dataset`?

File: cohere/client.py [hunk 4]
```diff
@@ -680,6 +681,109 @@ def rerank(
...
+        params = {
+            "name": name,
+            "type": dataset_type,
+            "keep_fields": keep_fields,
+            "optional_fields": optional_fields,
+        }
+        logger.warning("uploading file, starting validation...")
+        create_response = self._request(cohere.DATASET_URL, files=files, params=params)
+        logger.warning(f"{create_response['id']} was uploaded")
+        return self.get_dataset(id=create_response["id"])
```
[the remark this answers is not in the document]
Comment (lfayoux) [reply cohere/client.py:714]: the create response only contains an `id`

**Review by daniel-cohere — APPROVED** [review]

**Pull request merged**